macros/edr/tests/test_configuration/get_time_bucket.sql

```diff
@@ -0,0 +1,67 @@
+{% macro get_daily_time_bucket() %}
+  {% do return({"period": "day", "count": 1}) %}
+{% endmacro %}
+
+{% macro get_default_time_bucket() %}
+  {% do return(elementary.get_daily_time_bucket()) %}
+{% endmacro %}
+
+{% macro get_time_bucket(time_bucket, model_graph_node) %}
+    {%- set configured_time_bucket = elementary.get_test_argument('time_bucket', time_bucket, model_graph_node) %}
+    {%- do elementary.validate_time_bucket(configured_time_bucket) %}
+    {%- set default_time_bucket = elementary.get_default_time_bucket() %}
+
+    {%- if not configured_time_bucket %}
+        {{ return(default_time_bucket) }}
+    {%- elif configured_time_bucket and not configured_time_bucket.period %}
+        {%- do configured_time_bucket.update({"period": default_time_bucket.period }) -%}
+    {%- elif configured_time_bucket and not configured_time_bucket.count %}
+        {%- do configured_time_bucket.update({"count": default_time_bucket.count }) %}
+    {%- endif %}
+    {{ return(configured_time_bucket) }}
+{% endmacro %}
+
+
+{% macro validate_time_bucket(time_bucket) %}
+    {% if time_bucket %}
+        {%- if time_bucket is not mapping %}
+            {% do exceptions.raise_compiler_error(
+            "
+            Invalid time_bucket format. Expected format:
+
+               time_bucket:
+                 count: int
+                 period: string
+            ") %}
+        {%- endif %}
+        {%- if time_bucket is mapping %}
+            {%- set invalid_keys = [] %}
+            {%- set valid_keys = ['period', 'count'] %}
+            {%- for key, value in time_bucket.items() %}
```

Comment:
You should probably use `valid_keys` or preferable use the keys of the default time bucket.